Implement flatfile functionality for CNV and introduce cnvKompare

Kdreval opened the pull request:
This PR introduces the flatfile functionality for CNV to avoid reliance on the database. The useful feature here is a new `cnvKompare` function that will compare CNV data for patients with multiple time points and calculate the overall concordance of CNV + other comparison data. It can also handle the comparison of CNV between different tools if needed.

Kdreval commented:
This also addresses and resolves issue #19

lkhilton commented on R/utilities.R:
```diff
@@ -2910,3 +2910,310 @@ adjust_ploidy = function(this_seg,
 }
+
+
+#' This function will compare CNV data between samples with multiple time points. It can also handle same-sample comparison
```
This format will put this entire block of text into the title so when you look at the help for this function all these details are in large bold font at the top. The format should go like this: 

```
#' Sum of vector elements
#'
#' `sum` returns the sum of all the values present in its arguments.
#'
#' This is a generic function: methods can be defined for it directly
#' or via the [Summary()] group generic. For this to work properly,
#' the arguments `...` should be unnamed, and dispatch is on the
#' first argument.
```

(From https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html)

Perhaps while you fix this you could update the docstrings for `sanitize_maf_data`?

Kdreval replied:
Thanks, I'll fix this!

lkhilton reviewed (changes requested):
Awesome! Just needs an update to the docstrings.

Kdreval commented on R/utilities.R:
```diff
@@ -2910,3 +2910,310 @@ adjust_ploidy = function(this_seg,
+  # retrieve sample ids if only patient id is specified
+  if (missing(sample_ids)) {
+    sample_ids = get_gambl_metadata()
+    sample_ids = dplyr::filter(sample_ids, patient_id == {
```
Ah, this is syntax autoformatting! Have to be careful not to miss this the next time!

Kdreval commented:
Thanks for review Laura! I fixed these issues, and also added additional function to specify minimum similarity between cytobands to be considered concordant/discordant

lkhilton reviewed (approved):
Looks great! Adding percent concordance as a configurable parameter is a great addition.
